src/mgmt/cli_mgmt.c

```diff
@@ -849,43 +852,72 @@ dc_mgmt_get_pool_svc_ranks(struct dc_mgmt_sys *sys, const uuid_t puuid,
 	struct mgmt_pool_get_svcranks_in       *rpc_in;
 	struct mgmt_pool_get_svcranks_out      *rpc_out;
 	crt_opcode_t				opc;
-	int					rc;
+	int					i;
+	int					idx;
+	bool					success = false;
+	int					rc = 0;
+
+	/* NB: sy_psrs[] may have multiple entries even for single MS replica,
+	 * since there may be multiple ioservers there. Some of which may have
+	 * been stopped or faulted. May need to contact multiple ioservers.
+	 * Assumed: any MS replica ioserver can be contacted, even non-leaders.
+	 */
+	D_ASSERT(sys->sy_npsrs > 0);
+	idx = rand() % sys->sy_npsrs;
+	for (i = 0 ; i < sys->sy_npsrs; i++) {
+		srv_ep.ep_grp = sys->sy_group;
+		srv_ep.ep_rank = sys->sy_psrs[idx].rank;
+		srv_ep.ep_tag = daos_rpc_tag(DAOS_REQ_MGMT, 0);
+
+		rpc = NULL;
+		opc = DAOS_RPC_OPCODE(MGMT_POOL_GET_SVCRANKS, DAOS_MGMT_MODULE,
+				      DAOS_MGMT_VERSION);
+		rc = crt_req_create(daos_get_crt_ctx(), &srv_ep, opc, &rpc);
+		if (rc != 0) {
+			D_ERROR(DF_UUID ": crt_req_create() failed, "DF_RC "\n",
+				DP_UUID(puuid), DP_RC(rc));
+			return rc;
```

Comment:
Yes I think that's reasonable, including making the D_ERROR a D_DEBUG (in case req create works for a subsequent entry in the array). I'll probably remove some other statements from the loop that only need to be done once.

Reply:
I think D_ERROR would be fine here, as we would like to know more info on why crt_req_create() failed; it would be abnormal for it to not return 0, so error should be logged for us to investigate, but at the same time we can continue to pick next target